### PR TITLE
fix windows pythonnet warning unit tests

### LINKED ISSUE #247: [Embedding] embedding pythonnet warning test (test_warning_message) breaks on windows
when this test is run, it seems to modify the environment in such a way that embedding tests are subsequently broken. I think this is because the embedding tests all run in the same process, so whatever version of the CLR gets loaded first is the one that is always loaded.

### PR DESCRIPTION
**Pull request by koubaa**
I couldn't run the embedding unit tests locally anymore, and this change set fixes that. Fixes #247 

I also added some files to .gitignore that are artifacts of running remote session unit tests

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -119,6 +132,53 @@ def mke_app_reset(request):
...
+        # python executable inside the environment
+        python = os.path.join(venv_bin, exe_name)
+        # for convenience - the root directory of the pymechanical git repo
+        pymechanical_root = base.parent
+
+    test_env_object = TestEnv()
+
+    # Create virtual environment
+    subprocess.run([sys.executable, "-m", "venv", venv_dir], env=env_copy)
+    print(f"created virtual environment in {venv_dir}")
```
Comment (klmcadams) [inline tests/conftest.py:169]:
```suggestion
```

File: tests/conftest.py [hunk 2]
```diff
@@ -119,6 +132,53 @@ def mke_app_reset(request):
...
+
+    # Upgrade pip
+    cmdline = [test_env_object.python, "-m", "pip", "install", "-U", "pip"]
+    subprocess.check_call(cmdline, env=test_env_object.env)
+
+    yield test_env_object
+
+    print(f"\ndeleting virtual environment in {venv_dir}")
```
Comment (klmcadams) [inline tests/conftest.py:177]:
```suggestion
```

File: tests/conftest.py [hunk 3]
```diff
@@ -119,6 +132,53 @@ def mke_app_reset(request):
...
+    shutil.rmtree(venv_dir)
+    print(f"deleted virtual environment in {venv_dir}\n")
```
Comment (klmcadams) [inline tests/conftest.py:179]:
```suggestion
```

File: tests/conftest.py [hunk 4]
```diff
@@ -46,11 +51,19 @@ def pytest_collection_modifyitems(config, items):
     markexpr = config.option.markexpr
     if keywordexpr or markexpr:
         return  # command line has a -k or -m, let pytest handle it
+
+    # skip embedding tests unless the mark is specifiedskip_python_env
```
Comment (samigithub2022) [inline tests/conftest.py:55]: space missing in 'specifiedskip_python_env'

**Review by klmcadams — APPROVED** [review]

**Pull request merged**